docs/src/references/mappers/c8y-mapper.md

```diff
@@ -1325,3 +1325,104 @@ te/device/main///cmd/log_upload/<cmd_id>
+Crash itself (the mapper) with the hope of timely message delivery after the restart?
+
+### Out of order delivery on mapper restart
+
+The message delivery problem gets even worse during a mapper restart,
+as there could be hundreds of retained messages: registration messages, twin update messages and command metadata messages,
+all delivered in quick succession to the mapper.
+Due to the sheer volume of these messages, the chances of out of order delivery is also very high,
+leading to the mapper dropping a lot of critical messages, leaving the entity store in a partial/broken state.
+
+#### Initial solution thoughts
+
+Handling this issue on a restart is easier, by using a special MQTT subscriber just on starup,
+which will subscribe exclusively to the registration messages and process them,
+before starting the regular message processing.
+
+But, if we have a solution for the live message processing problem, the same can be used here as well,
+without having to invent another solution for the restart case.
+
+### Auto-registration interfering with explicit registration
+
+When auto-registration is enabled, and if some users use explicit registration along with it,
+the above mentioned out-of-order message delivery can lead to entities getting created with the wrong metadata,
+like the `external-id` derived by the auto-registration logic,
+even if the user had explicitly provided those values in their registration message.
+
+This case is illustrated in the following diagram:
+
+```mermaid
+sequenceDiagram
+    participant Child device
+    participant MQTT Broker
+    participant Tedge Mapper
+    
+    Child device ->> MQTT Broker: Child device registration message to `te/device/plc1//` `{ "id": "desired-xid" }`
+    Child device ->> MQTT Broker: Telemetry message to `te/device/plc1///m/temperature`
+    MQTT Broker ->> Tedge Mapper: Telemetry message to `te/device/plc1///m/temperature`
+    Tedge Mapper ->> Tedge Mapper: Auto-register the `plc1` device with auto-generated `external-id`
+    MQTT Broker ->> Tedge Mapper: Child device registration message to `te/floor1/plc1//` `{ "id": "desired-xid" }`
+    Tedge Mapper ->> Tedge Mapper: Ignore the registration message as the entity is already registered
+```
+
+Here the end-result is even more catastrophic, as the user expects a device with external-id: `desired-xid`,
+but the mapper has created one with the auto-generated external-id, which can't be changed afterwards.
+
+#### Initial solution thoughts
+
+One obvious solution to avoid this problem is to force the users to disable auto-registration,
+when they intend to use explicit registration even for a single entity.
+But the downside with that approach is that we force them to use explicit registration for everything,
+even the entities for which auto-registration would have been perfectly fine,
+just because a few entities required some customizations via explicit registration.
```

Comment:
I agree too. Anyone opting for a custom topic scheme is more likely to stick to that scheme for his entire fleet and not just a few devices. There was a typo in the above sequence diagram where the 5th message from the broker to the mapper was wrongly written as `te/floor1/plc1//` instead of `te/device/plc1//`, which may have caused the confusion. Corrected now.

But, unfortunately the issue highlighted here is not just limited to custom topic schemes but even for the default topic scheme in cases where an explicit registration is mandatory (nested child devices or when that customer wants to control the `id` of all devices in his fleet, without relying on the auto-generated IDs).